lib/index.ts

```diff
@@ -29,13 +29,15 @@ export const generateRemoteUrl = (service: string) => window.location.protocol +
 /**
  * Get the base path for the given OCS API service
  *
- * @param {string} service name
+ * @param {string} url OCS API service url
  * @param {int} version OCS API version
- * @return {string} OCS API base path
+ * @param {object} params parameters to be replaced into the service url
+ * @param {UrlOptions} options options for the parameter replacement
+ * @return {string} Absolute path for the OCS URL
  */
-export const generateOcsUrl = (service: string, version: Number) => {
+export const generateOcsUrl = (url: string, version: Number, params?: object, options?: UrlOptions) => {
```

Comment:
So basically everyone has to touch their stuff anyway, we could also turn around the argument order? For me it always felt weird that I first give the service and then the version, while the url is version and then service?

Opinions? @ChristophWurst @skjnldsv

Reply:
I'm fine with both :shrug: